smart_kit/utils/diff.py

```diff
@@ -20,8 +20,10 @@ def __init__(self):
     @classmethod
     def partial_diff(cls, expected: Any, actual: Any, path="", dict_check_does_not_expected=False) -> Diff:
         diff = cls()
-        if expected != actual and \
-                not (isinstance(actual, str) and isinstance(expected, str) and re.fullmatch(expected, actual)):
+        if (not isinstance(actual, type(expected)) or
+           (isinstance(actual, str) and isinstance(expected, str) and
```

Comment:
```suggestion
           (isinstance(actual, str) and
```
Из предыдущего экспрешна следует, что типы совпадают